### ISSUE TITLE
json vs yaml source?

### ISSUE THREAD

**Issue by bdarcus** [citation-style-language/data-parsing#1]
Right now, machine-readable source files are json. But we could do yaml, and run the json conversion when publishing.

**Comment by customcommander** [comment]
I'm definitely missing lots of background information but I thought I would leave a comment:

While I do like the simplicity of authoring YAML files, I find that using JSON allows other useful tools such as jq and json-schema. Most programming languages offer JSON support out of the box too.

**Comment by bdarcus** [comment]
If we did YAML, it would only be for the source files; they would be converted to JSON when GitHub pages runs.

The purpose would be to make it easier for the people creating the content.

The downside of that, though, is people couldn't just use them from a local repo without first converting.

This json is simple enough, however, that YAML may not be a significant enough benefit on the authoring side.

**Comment by bdarcus** [comment]
BTW, @customcommander - this was a test of an idea to address the issue I linked from above. We've modified the input to more clearly express the modelling expectations for CSL input data, so this would just help developers know how to generate that data, when their source are strings. If any more input, let me know, either here, or on that linked issue on the main schema repo.

https://github.com/citation-style-language/schema/issues/324

Also, I wasn't familiar with `jq`, but in the same way many tools work with yaml, there's `yq`, that is a wrapper that will process yaml.

``` console
❯ yq '.examples[1].output' ../yaml/titles.yaml
{
  "main": "Whose Music?",
  "sub": [
    "A Sociology of Musical Language"
  ]
}
```
